vital/config/tests/test_config_block_io.cxx

```diff
@@ -374,18 +354,14 @@ TEST_F( config_block_io, write_config_simple_success )
 
   print_config( orig_config, false, "ConfigBlock for writing" );
 
-  char output_path_1[128] = "test_config_output_1.conf.XXXXXX";
```

Comment:
Honestly, I don't understand what the old code was doing and why it needed to open and close a file.  I'm okay with this change as long as it doesn't remove some important aspect of the test.  I think @linus-sherrill should comment on this one.

Reply:
Ah, right... the old code was trying to avoid using "dangerous" [`tempnam`](https://linux.die.net/man/3/tempnam). However, the way to avoid what makes `tempnam` dangerous is to use [`mkstemp`](https://linux.die.net/man/3/mkstemp) instead and write directly to the FD, which `write_config_file` doesn't support. The "work-around" was using that to avoid the warning, which also opens the file, then close the file (to not leak the FD), and then write to the path with `write_config_file`. This effectively circumvented the added security of `mkstemp`.

Reply:
@linus-sherrill can you confirm that you are okay with this change?  If so then this branch is ready to merge.